system_zero/src/registers/alu.rs

```diff
@@ -89,4 +93,34 @@ pub(crate) const COL_DIV_RANGE_CHECKED_TMP_0: usize = super::range_check_16::col
 /// The second 16-bit chunk of a temporary value (divisor - remainder - 1).
 pub(crate) const COL_DIV_RANGE_CHECKED_TMP_1: usize = super::range_check_16::col_rc_16_input(5);
 
+///
+/// Bitwise logic operations
+///
+
+/// Bit decomposition of 64-bit values, as 32-bit low and high halves.
+
+const fn gen_bitop_32bit_input_regs(start: usize) -> [usize; 32] {
+    let mut regs = [0usize; 32];
+    let mut i = 0;
+    while i < 32 {
+        regs[i] = shared_col(start + i);
+        i += 1;
+    }
+    regs
+}
+
+pub(crate) const COL_BIT_DECOMP_INPUT_A_LO_BIN_REGS: [usize; 32] = gen_bitop_32bit_input_regs(0);
+pub(crate) const COL_BIT_DECOMP_INPUT_A_HI_BIN_REGS: [usize; 32] = gen_bitop_32bit_input_regs(32);
+pub(crate) const COL_BIT_DECOMP_INPUT_B_LO_BIN_REGS: [usize; 32] = gen_bitop_32bit_input_regs(64);
+pub(crate) const COL_BIT_DECOMP_INPUT_B_HI_BIN_REGS: [usize; 32] = gen_bitop_32bit_input_regs(96);
+
+/// The first 16-bit chunk of the output, based on little-endian ordering.
+pub(crate) const COL_BITOP_OUTPUT_0: usize = super::range_check_16::col_rc_16_input(0);
```

Comment:
Hm I would have expected outputs to be in 32-bit chunks. For mul etc. we used 16-bit output chunks since they needed to be range-checked, but here the outputs' range should be enforced by the constraints, I think.

I suppose there's no harm in this though, since it's using registers that would otherwise be unused for bit ops. It's nice not to need extra registers.

In the future, maybe we can write outputs like this directly to the memory bus (though it's not implemented yet).

Reply:
I've updated this to return two 32-bit chunks, taking two extra registers (total 130 for bit ops). We can always change back if we want to save those two extra registers.